bigquery/src/main/scala/com/yahoo/maha/executor/bigquery/BigqueryQueryExecutor.scala

```diff
@@ -0,0 +1,227 @@
+import com.google.cloud.ServiceOptions
+import com.google.cloud.bigquery.BigQuery
+import com.google.cloud.bigquery.BigQueryOptions
+import com.google.cloud.bigquery.JobId
+import com.google.cloud.bigquery.JobInfo
+import com.google.cloud.bigquery.QueryJobConfiguration
+import com.google.cloud.http.HttpTransportOptions
+import com.yahoo.maha.core._
+import com.yahoo.maha.core.query._
+import com.yahoo.maha.core.request.{Parameter, RequestIdValue}
+import grizzled.slf4j.Logging
+import java.io.FileInputStream
+import java.util.UUID
+import org.apache.http.HttpHost
+import org.apache.http.auth.{AuthScope, UsernamePasswordCredentials}
+import org.apache.http.impl.client.{
+  BasicCredentialsProvider,
+  DefaultHttpRequestRetryHandler,
+  ProxyAuthenticationStrategy
+}
+import org.apache.http.impl.conn.DefaultProxyRoutePlanner
+import scala.util.Try
+
+private case class ProxyCredentials(
+  @JsonProperty("proxy_user") userName: String,
+  @JsonProperty("proxy_pass") password: String
+)
+
+case class BigqueryQueryExecutorConfig(
+  gcpCredentialsFilePath: String,
+  gcpProjectId: String,
+  enableProxy: Boolean,
+  proxyCredentialsFilePath: Option[String],
+  proxyHost: Option[String],
+  proxyPort: Option[String],
+  retries: Int
+) extends Logging {
+
+  private val mapper = {
+    val om = new ObjectMapper(new YAMLFactory()) with ScalaObjectMapper
+    om.registerModule(DefaultScalaModule)
+    om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    om
+  }
+
+  def buildBigqueryClient(): BigQuery = {
+    val builder = BigQueryOptions
+      .newBuilder()
+      .setProjectId(gcpProjectId)
+      .setRetrySettings(
+        ServiceOptions
+          .getDefaultRetrySettings()
+          .toBuilder()
+          .setMaxAttempts(retries)
+          .build()
+      )
+
+    val builderWithCredentials = if (!gcpCredentialsFilePath.isEmpty) {
+      val fileInputStream = new FileInputStream(gcpCredentialsFilePath)
+      try {
+        if (enableProxy && proxyCredentialsFilePath.isDefined && proxyHost.isDefined && proxyHost.isDefined) {
+          val proxyTransport = googleApiHttpTransportFactory()
+          builder
+            .setCredentials(ServiceAccountCredentials.fromStream(fileInputStream, proxyTransport))
+            .setTransportOptions(
+              HttpTransportOptions
+                .newBuilder()
+                .setHttpTransportFactory(proxyTransport)
+                .build()
+            )
+        } else {
+          builder.setCredentials(ServiceAccountCredentials.fromStream(fileInputStream))
+        }
+      } finally {
+        fileInputStream.close()
+      }
+    } else {
+      builder
+    }
+
+    builderWithCredentials
+      .build()
+      .getService
+  }
+
+  private[this] def googleApiHttpTransportFactory(): HttpTransportFactory = {
+    info(s"Setting up proxied Google API Http Transport")
+    val fileInputStream = new FileInputStream(proxyCredentialsFilePath.get)
```

Comment:
This looks like an unsafe call to get(), you could do something like this:
```
private[this] def googleApiHttpTransportFactory(): Option[HttpTransportFactory] = {
  if (enableProxy && proxyCredentialsFilePath.isDefined && proxyHost.isDefined && proxyHost.isDefined) {
    info(s"Setting up proxied Google API Http Transport")
    val fileInputStream = new FileInputStream(proxyCredentialsFilePath.get)
    try {
      ...
      return Option(mHttpTransport)
    }
  } else None
```

Reply:
Then you can change line 73 if block to something like this:
```
googleApiHttpTransportFactory().fold(builder.setCredentials(ServiceAccountCredentials.fromStream(fileInputStream))) {
  proxyTransport =>
    ...
}
```

Reply:
Done